### DISCUSSION TITLE
How to type set action without setXXX ?

### DISCUSSION THREAD

**Discussion by vaynevayne** [pmndrs/zustand#2599] [Q&A] [answered]
```jsx

type Actions = {
  set: () => void //  ???
  reset: () => void
}
const useLocalStore = create<State & Actions>()(
  persist(
    (set, get) => ({
      ...initialState,
      set,
      reset: () => {
        set(initialState)
      },
    }),
    {
      name: 'log-storage',
      storage: createJSONStorage(() => localStorage),
    },
  ),
)
```

https://www.typescriptlang.org/play/?#code/FASwtgDg9gTgLgAgN4IMYwKYEM4YQXwQDMYowEByALwFcBnOLAOwBMLRJZEV1tcApAMoB5AHKC4sLAHMMAGgQQMMOiAYFipctXqNWAejAgWLADYYA7lkztg+-QhYYiIJnjgBPJXWKwEDHDwANyxTGgwfZhYELFQ4ECgmHzoMCGtA0w9gTyUECUCEAF5kYAQEUyhpAFEmLAAjcxYALgQ6qChzZmB8YGyvPABBOISkopKylLgWgAoASiKAPgQgqGNShExJmfnCpZW1nrsHJxc3BDgACzxXEHjQ-0ZcYFRE9Ru703zcFq+8YqR1hVqrUGhhmsRQik5N1evY0JgCgxYBhnq9EPQMAAZKCoUISZFjXiBAA8vwQADIEEN4q8FnNpuslCo1HAGWUytNJgpZHAdktpgD2eyAHSi94gPGPeTrIVcmXszYYKYIOaLcZC9mTabiyWBWbysr4aFC-CzY3swUa2pgDAtChAgC0SJgMgwFHNsskLtkLSJAhE4i9rumqt25RxkqksjN8qN6xj+uAGAAHtB4I5nFgaKZ0SlsbjPl6UUA

**Comment by dai-shi** [comment, marked as answer]
It depends on how much api you want to expose, but for example this works: https://tsplay.dev/Nnj2vN

**Reply by vaynevayne** [reply]
Does setaction also support function params ? 
https://www.typescriptlang.org/play/?#code/FASwtgDg9gTgLgAgN4IMYwKYEM4YQXwQDMYowEByALwFcBnOLAOwBMLRJZEV1tcApAMoB5AHKC4sLAHMMAGgQQMMOiAYFipctXqNWAejAgWLADYYA7lkztg+-QhYYiIJnjgBPJXWKwEDHDwANyxTGgwfZhYELFQ4ECgmHzoMCGtA0w9gTyUECUCEAF5kYAQEUyhpAFEmLAAjcxYALgQ6qChzZmB8YGyvPABBOISkopKylLgWgAo0+BBQloAFa3jQgB583AA+BAAfBGnZzCCmrYwASkLd84uLot2gqGNShExJmfvrhCeXnrsHE4XG4EHAABZ4VwgNamfyMXDAVCJdRQmHnFrnMZIV4Vaq1BoYZrEUIpOTdXr2NCYAoMWAYRHIxD0DAAGSgqFCEjpY14gU28LwADIEEN4sjttMLtNXkoVGo4NKymVppMFLI4F9dtNsUqlQA6A2ohamc5k3UTDBwM3m96Wz4Pcbmi0Ko2cgUXV66-DWsr4C4+x262pgDAtCi4gC0tJgMgwFADE0kMdkLV5AhE4iTsaOmvK7LdUlk-s9BGtxY9wAwAA9oPBHM4sDRTEyUmyOSak-SgA

**Reply by dai-shi** [reply]
Yeah, it can be `Partial<>`. https://tsplay.dev/mpk2MN

**Reply by vaynevayne** [reply]
xi  thanks
